icicle/tests/test_hash_api.cpp

```diff
@@ -391,48 +560,209 @@ TEST_F(HashApiTest, MerkleTreeBasic)
   output_store_min_layer = output_store_min_layer & 3; // Ensure index is in a valid 0-3 range
   ICICLE_LOG_DEBUG << "Min store layer:\t" << output_store_min_layer;
 
-  auto prover_tree = MerkleTree::create(hashes, sizeof(uint32_t), output_store_min_layer);
-  auto verifier_tree = MerkleTree::create(hashes, sizeof(uint32_t), output_store_min_layer);
+  test_merkle_tree(hashes, config, output_store_min_layer, nof_leaves, leaves);
+}
 
-  // build tree
-  ICICLE_CHECK(prover_tree.build(leaves, nof_leaves, config));
-  assert_valid_tree<uint32_t>(prover_tree, nof_leaves, leaves, hashes, config);
+TEST_F(HashApiTest, MerkleTreeZeroPadding)
```

Comment:
you need to test it for all devices with a loop like MerkleTreeMixMediumSize
This is true for all tests except for where you use the dummy HashSum since it's only CPU

Reply:
Again I think it is not yet implemented on GPU